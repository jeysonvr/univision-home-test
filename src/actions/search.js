import { types } from "../types/types";

export const setKeyword = (keyword, offset) => {
    
    return (dispatch) => {
        dispatch( {
            type: types.setParameters,
            payload: {
                keyword: keyword,
                offset: offset
            }
        });
    }
}

export const setFullImage = (url, title) => {
    
    return (dispatch) => {
        dispatch( {
            type: types.setFullImage,
            payload: {
                imgURL: url,
                imgTitle: title
            }
        });
    }
}

export const setScrollSlider = (scroll) => {

    return (dispatch) => {
        dispatch( {
            type: types.setScrollSlider,
            payload: {
                scroll: scroll
            }
        });
    }

}