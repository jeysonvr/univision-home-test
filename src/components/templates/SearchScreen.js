import React, { useEffect } from 'react'
import { useDispatch } from "react-redux";
import { setKeyword, setScrollSlider } from '../../actions/search';
import { SearchForm } from '../organisms/SearchForm';
import { SearchResult } from '../organisms/SearchResult';


export const SearchScreen = () => {
    
    const dispatch = useDispatch();

    useEffect(() => {
        dispatch( setKeyword('', 0) );
    }, []);

    const handleSearch = (keyword) => {
        dispatch( setKeyword(keyword, 0) );
        dispatch( setScrollSlider(0) );
    }
    
    return (
        <div className="container">
            <div className="row">
                <h3>Search your favorite gifts</h3>
                <hr />
            </div>
            <SearchForm
                handleSearch={ handleSearch }
            />
            <SearchResult />
        </div>
    )
}
