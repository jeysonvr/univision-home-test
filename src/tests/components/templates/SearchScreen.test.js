import React from 'react';
import { mount } from "enzyme";
import { Provider } from "react-redux";

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom';

import { SearchScreen } from "../../../components/templates/SearchScreen";
import { setKeyword, setScrollSlider } from "../../../actions/search";

jest.mock('../../../actions/search', () => ({
    setScrollSlider: jest.fn(),
    setKeyword: jest.fn()
}))

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {
    search: {
        keyword: 'test', 
        offset: 0, 
        imgURL: 'a', 
        imgTitle: 'b'
    }
};
let store = mockStore(initialState);
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={ store }>
        <SearchScreen /> 
    </Provider>
);

describe('Test in <SearchScreen />', () => {

    beforeEach( () => {
        store = mockStore(initialState);
        jest.clearAllMocks();
    })

    test('should show component correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    })
    
    test('should trigger action handleSearch', () => {
        
        wrapper.find('.btn-primary').prop('onClick')({preventDefault(){}});
        expect( setKeyword ).toHaveBeenCalled();
        expect( setScrollSlider ).toHaveBeenCalled();
    })
    

})