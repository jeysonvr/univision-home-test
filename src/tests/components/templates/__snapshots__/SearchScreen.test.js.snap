// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <SearchScreen /> should show component correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <h3>
      Search your favorite gifs
    </h3>
    <hr />
  </div>
  <form
    className="row m-2"
    onSubmit={[Function]}
  >
    <label
      className="col-2 col-sm-10 col-md-2"
    >
      Keyword: 
    </label>
    <input
      autoComplete="off"
      className="col-12 col-sm-10 col-md-6"
      name="keyword"
      onChange={[Function]}
      placeholder="Search ..."
      style={
        Object {
          "marginRight": "10px",
        }
      }
      type="text"
      value=""
    />
    <button
      className="btn btn-primary col-sm-10 col-md-3 mt-1"
      onClick={[Function]}
    >
      <i
        className=""
      />
      Search
    </button>
  </form>
  Array [
    Array [
      <p
        className="card"
      >
        Loading
      </p>,
      <div
        className="alert alert-primary m-5"
        role="alert"
      >
        Search something ...
      </div>,
    ],
    Array [
      <div
        className="modal animate__animated animate__fadeIn"
        style={
          Object {
            "display": "block",
          }
        }
        tabIndex="-1"
      >
        <div
          className="modal-dialog modal-dialog-centered"
        >
          <div
            className="modal-content text-center"
          >
            <div
              className="modal-header"
            >
              <h5
                className="modal-title"
              >
                b
              </h5>
              <button
                className="btn button__close"
                onClick={[Function]}
              >
                <i
                  className="fas fa-times-circle fa-2x"
                />
              </button>
            </div>
            <div
              className="modal-body card"
            >
              <img
                alt="b"
                className="img-fluid"
                src="a"
              />
            </div>
          </div>
        </div>
      </div>,
      <div
        className="modal-backdrop fade show"
      />,
    ],
  ]
</div>
`;
